### DISCUSSION TITLE
Deploying Remix on Kubernetes

### DISCUSSION THREAD

**Discussion by chikamichi** [remix-run/remix#5598] [Show and tell / tips]
Hey there,

I published [a tutorial on how to deploy a Remix app (w/ Express Server) on Kubernetes](https://kaibun.net/articles/2023-02-25-deploying-remix-to-kubernetes) 💡

It might prove useful to some, in the absence of an official documentation on the subject.

**Comment by sbaechler** [comment]
@chikamichi I cannot access the link. It looks like kaibun.net is not reachable.

**Comment by chikamichi** [comment]
Hi, I had to take it down temporarily, couldn’t afford running that server atm. Hopefully I’ll host that website again somewhere else.

In the meantime, I can provide you with the article’s .md, while the code repo is [here](https://github.com/kaibun/my-remix-app-on-kubernetes).

**Comment by chikamichi** [comment]
To get straight to the point, I hosted the raw .md at https://github.com/kaibun/my-remix-app-on-kubernetes/blob/main/article.md so it lacks images, code highlighting and whatnots, but you don’t need them to benefit from the actual tech content.

**Comment by chikamichi** [comment]
@sbaechler the article is back online: https://kaibun.net/articles/2023-02-25-deploying-remix-to-kubernetes
I’m not hosting the example app though, as I went with a simple "native" Nginx setup and not a full-fledged k8s cluster. No big deal though, that app serves no purpose beyond demonstrating that "it works".

**Reply by sbaechler** [reply]
Thanks for the writeup. This will come in handy. I also took some inspiration from the [docker file of the epic stack](https://github.com/epicweb-dev/epic-stack/blob/main/other/Dockerfile). KCD pruned the dev dependencies after build which results in smaller containers.
